js/state-from-store-mixin.jsx

```diff
@@ -140,16 +150,11 @@ var StateFromStore = function(stateDescriptors) {
                 changeListeners: {}
             };
 
-            var initialState = {};
-            Object.keys(stateDescriptors).forEach(function(stateKey) {
-                var stateData = fetchNewStateData(
-                    this, stateKey, false).stateData;
-                initialState[stateKey] = stateData;
-            }, this);
-            return initialState;
+            return getStoreState(this);
         },
 
         componentDidMount: function() {
+            this.setState(getStoreState(this));
```

Comment:
And the rerender _should_ happen, or the UI won't match the state
On Thu, May 14, 2015 at 3:37 PM Zach Gotsch <orbipedis@gmail.com> wrote:

> You can't set state on unmounted components
> On Thu, May 14, 2015 at 3:04 PM Emily Eisenberg <notifications@github.com>
> wrote:
>
>> In js/state-from-store-mixin.jsx
>> <https://github.com/Khan/react-components/pull/39#discussion_r30351805>:
>>
>> >          },
>> >
>> >          componentDidMount: function() {
>> > +            this.setState(getStoreState(this));
>>
>> Nooo don't do this, it causes a re-render if the state changes! Maybe we
>> could move this to componentWillMount, so we don't cause this to happen?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Khan/react-components/pull/39/files#r30351805>.
>>
>